OTA-905: ipi install: use integration cincinnati in CI clusters, where possible

petr-muller opened the pull request:
This PR is a part of the same effort as https://github.com/openshift/release/pull/40703

OTA maintains staging (identical to production) and integration (running on engineering candidate OCP clusters) Cincinnati instances. We are searching for traffic that we could route to especially the integration one, so that we can find possible problems with engineering candidate early.

All the instances are serving identical data (up to some minimal skew coming from when individual instance scrape their source data) so we should be able to easily use the integration instance in CI clusters that are running a released OCP version. Most CI clusters are not running such versions, and these need to *not* query OSUS, otherwise
they would trip an alert, causing noise in CI jobs. CI clusters are not querying OSUS since https://github.com/openshift/release/pull/8631

We can enhance the logic to validate whether the version we are going to install is known to OSUS. If it is, we know we are installing a published OCP version and it is safe to let the cluster query OSUS. We still clear the channel to prevent the cluster from querying OSUS if we the version we install is not known to OSUS.

Changing `ipi-install-install`, this has a huge blast radius with these three potential failure cases:

1. Non-zero exit code of my bash will cause steps to fail. I ran a ton of rehearsals.
2. Integration Cincinnati breaks and clusters will start to alert about failing to query OSUS. That's the point of this change. Integration OSUS runs on Engineering Candidate OCP cluster and if ECs break real-world workloads, we want to know that.
3. Some clusters will query for updates when they should not. There can be corner cases that I missed, like some rogue update jobs that do not use `ipi-install-install-stableinitial` steps. I plan to monitor search.ci for an uptick of [jobs where `CannotRetrieveUpdates` alert fires](https://search.ci.openshift.org/?search=CannotRetrieveUpdates&maxAge=12h&context=1&type=junit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job)

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
/pj-test all

petr-muller commented:
/test all

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator

petr-muller commented:
This is what I wanted to see (from https://console-openshift-console.apps.build05.l9oh.p1.openshiftapps.com/k8s/ns/ci-op-zvddhzch/pods/maistra-envoy-unit-2-4-ipi-install-install/logs)

```
Release payload version: 4.10.62
Original channel from CVO manifest: stable-4.10
Matching candidate channel: candidate-4.10
Version 4.10.62 is available in candidate-4.10: cluster can query OSUS
```

wking commented on ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:
```diff
@@ -427,7 +427,32 @@ aws|aws-arm64|aws-usgov)
     ;;
 esac
 
-sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
```
I dunno why folks have copy/pasted this around so much, but:

```console
$ git --no-pager grep cvo-overrides.yaml ci-operator/step-registry/
ci-operator/step-registry/ipi/install/install/aws/ipi-install-install-aws-commands.sh:sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
ci-operator/step-registry/ipi/install/libvirt/install/ipi-install-libvirt-install-commands.sh:sed -i '/^  channel:/d' ${dir}/manifests/cvo-overrides.yaml
ci-operator/step-registry/ipi/install/powervs/install/ipi-install-powervs-install-commands.sh:sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
ci-operator/step-registry/upi/conf/vsphere/platform-external/upi-conf-vsphere-platform-external-commands.sh:sed -i '/^  channel:/d' "manifests/cvo-overrides.yaml"
ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh:sed -i '/^  channel:/d' "manifests/cvo-overrides.yaml"
ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh:sed -i '/^  channel:/d' "manifests/cvo-overrides.yaml"
ci-operator/step-registry/upi/install/aws/cluster/upi-install-aws-cluster-commands.sh:sed -i '/^  channel:/d' ${ARTIFACT_DIR}/installer/manifests/cvo-overrides.yaml
ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh:sed -i '/^  channel:/d' manifests/cvo-overrides.yaml
```

I'm completely fine leaving those other steps alone, and letting their maintainers continue to copy/paste or find some new way to DRY things up, but wanted to point out the issue in case other folks wanted to do something more proactive.

petr-muller replied:
I think I will try to update them all, but I will do that in separate PRs otherwise the rehearsals will be pretty hard to track.

wking commented on ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:
```diff
@@ -427,7 +427,32 @@ aws|aws-arm64|aws-usgov)
+    candidate_channel="$(echo "${channel}" | sed -E 's/(stable|eus|fast)/candidate/')"
+    echo "Matching candidate channel: ${candidate_channel}"
+
+    if curl --silent "https://api.integration.openshift.com/api/upgrades_info/v1/graph?channel=$candidate_channel" | grep --quiet '"version":"'"$payload_version"'"'; then
```
did we want to set `arch` too?  Or are we comfortable assuming that if the version is in channels for `amd64` it will exist for whatever arch this release happens to be?

petr-muller replied:
I think we should do that, but I have trouble finding a good source for that string. It seems that `multi` payloads have a `.metadata.metadata['release.openshift.io/architecture']` key but other do not, but there's also `.config.architecture`  which is correct everywhere except multi...:

```go
$ for arch in x86_64 ppc64le aarch64 multi
      set meta (oc adm release info quay.io/openshift-release-dev/ocp-release:4.13.6-$arch -o "jsonpath={.metadata.metadata.release\.openshift\.io/architecture}")
      echo (string pad -r -w 8 $arch)": .metadata.metadata['release.openshift.io/architecture']=$meta"
      set cfg (oc adm release info quay.io/openshift-release-dev/ocp-release:4.13.6-$arch -o "jsonpath={.config.architecture}")
      echo (string pad -r -w 8 $arch)": .config.architecture=$cfg"
  end

x86_64  : .metadata.metadata['release.openshift.io/architecture']=
x86_64  : .config.architecture=amd64
ppc64le : .metadata.metadata['release.openshift.io/architecture']=
ppc64le : .config.architecture=ppc64le
aarch64 : .metadata.metadata['release.openshift.io/architecture']=
aarch64 : .config.architecture=arm64
multi   : .metadata.metadata['release.openshift.io/architecture']=multi
multi   : .config.architecture=amd64
```

I guess OSUS needs some data when it scrapes, I guess I should look at what it does...

petr-muller replied:
Done

wking commented on ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:
```diff
@@ -427,7 +427,32 @@ aws|aws-arm64|aws-usgov)
+    echo "Matching candidate channel: ${candidate_channel}"
+
+    if curl --silent "https://api.integration.openshift.com/api/upgrades_info/v1/graph?channel=$candidate_channel" | grep --quiet '"version":"'"$payload_version"'"'; then
+        echo "Version ${payload_version} is available in ${candidate_channel}: cluster can query OSUS"
```
nit: I'd rather spell out "the OpenShift Update Service", to help out folks who haven't internalized "OSUS" (probably most readers ;) )

wking commented on ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:
```diff
@@ -427,7 +427,32 @@ aws|aws-arm64|aws-usgov)
+    else
+        echo "Version ${payload_version} is not available in ${candidate_channel}: cluster cannot query OSUS, clearing the channel"
+        sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
+        echo '  upstream: https://api.integration.openshift.com/api/upgrades_info/v1/graph' >> "${dir}/manifests/cvo-overrides.yaml"
```
why set `upstream` when the `channel` is cleared, and the target release isn't available in the integration update service responses?  Can we drop this line?

wking commented on ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:
```diff
@@ -427,7 +427,32 @@ aws|aws-arm64|aws-usgov)
 esac
 
-sed -i '/^  channel:/d' "${dir}/manifests/cvo-overrides.yaml"
+function setup-cvo() {
```
nit: this function can get shifted up towards the top of the file to avoid distracting folks reading the main install portion and trying to understand the big-picture flow.  And a name like `set-cluster-version-spec-update-service` might be easier for folks skimming past the call site to understand what's happening.

petr-muller commented:
/test build04-dry

petr-muller commented:
I cleaned up the change, added proper commit / pr description and I want to see some rehearsals. 

/hold
I still need to address Trevor's comment about querying OSUS with `arch` parameter (https://github.com/openshift/release/pull/40711#discussion_r1244189940)

petr-muller commented:
> A total of 11049 jobs have been affected by this change.
I fear nothing!

/pj-rehearse

petr-muller commented:
/pj-rehearse pull-ci-3scale-3scale-operator-master-test-e2e

petr-muller commented:
/pj-rehearse /pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e

petr-muller commented:
The auto-selected rehearsals are all using ephemeral release payload images, where we cannot query OSUS and have to channel clear:

[example 1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-openshift-coredns-master-e2e-aws-ovn/1682094688531648512/artifacts/e2e-aws-ovn/ipi-install-install/build-log.txt) [example 2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e/1682094688284184576/artifacts/openshift-e2e/ipi-install-install/build-log.txt)
```
Release payload version: 4.14.0-0.ci.test-2023-07-20-183529-ci-op-hjk4xqp1-latest
Original channel from CVO manifest: stable-4.14
Matching candidate channel: candidate-4.14
Version 4.14.0-0.ci.test-2023-07-20-183529-ci-op-hjk4xqp1-latest is not available in candidate-4.14: cluster cannot query OpenShift Update Service (OSUS/Cincinnati), clearing the channel
```

[clusterversion 1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-openshift-coredns-master-e2e-aws-ovn/1682094688531648512/artifacts/e2e-aws-ovn/gather-extra/artifacts/clusterversion.json) [clusterversion 2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e/1682094688284184576/artifacts/openshift-e2e/gather-extra/artifacts/clusterversion.json)
```json
            "spec": {
                "clusterID": "3151189b-f0a0-470d-b528-84e2100f8409"
            },
```

petr-muller commented:
All released OCP rehearsals look good as well

[example 1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e/1682419497660583936/artifacts/odh-notebook-controller-e2e/ipi-install-install/build-log.txt) [example 2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-3scale-3scale-operator-master-test-e2e/1682417518452412416/artifacts/test-e2e/ipi-install-install/build-log.txt) [example 3](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4/1682419497589280768/artifacts/maistra-envoy-unit-2-4/ipi-install-install/build-log.txt) [example 4](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic/1682419497706721280/artifacts/e2e-openshift-test-scenario-1-periodic/ipi-install-install/build-log.txt)

```
Release payload version: 4.10.63
Original channel from CVO manifest: stable-4.10
Matching candidate channel: candidate-4.10
Version 4.10.63 is available in candidate-4.10: cluster can query OpenShift Update Service (OSUS/Cincinnati)
```

[clusterversion 1](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e/1682419497660583936/artifacts/odh-notebook-controller-e2e/gather-extra/artifacts/clusterversion.json) [clusterversion 2](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-3scale-3scale-operator-master-test-e2e/1682417518452412416/artifacts/test-e2e/gather-extra/artifacts/clusterversion.json) [clusterversion 3](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4/1682419497589280768/artifacts/maistra-envoy-unit-2-4/gather-extra/artifacts/clusterversion.json) [clusterversion 4](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic/1682419497706721280/artifacts/e2e-openshift-test-scenario-1-periodic/gather-extra/artifacts/clusterversion.json)

```json
            "spec": {
                "channel": "stable-4.10",
                "clusterID": "705ac5fb-5402-499e-9839-a09616b7150f",
                "upstream": "https://api.integration.openshift.com/api/upgrades_info/graph"
            },
```

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e  pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade

petr-muller commented:
- [pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4/1687529131320807424) looks good: uses OSUS, candidate channel, no suspicious failures or alerts
- [pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e/1687529131379527680): failed to execute, will rerun
- [periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade/1687529131543105536): uses osus before update and candidate channel, but goes into `RetrievedUpdate=False` after the update, this needs to be fixed
- [periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade/1687529131438247936): failed to install (`Cluster operators kube-controller-manager, kube-scheduler are degraded`), does not seem related. Uses OSUS and candidate channel.
- [pull-ci-3scale-3scale-operator-master-test-e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-3scale-3scale-operator-master-test-e2e/1687529131136258048): looks good: uses OSUS, candidate channel, no suspicious failures or alerts
- [pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e/1687529131266281472): looks good, (correctly) does not use OSUS, channel cleared, no issues
- [pull-ci-openshift-coredns-master-e2e-aws-ovn](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-pull-ci-openshift-coredns-master-e2e-aws-ovn/1687529131203366912): looks good, (correctly) does not use OSUS, channel cleared, no issues
- [periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40711/rehearse-40711-periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic/1687529131484385280): looks good: uses OSUS, candidate channel, no suspicious failures or alerts

Next steps:
- [ ] Find a good multiarch payload-using job to rehearse
- [ ] Improve logging, some loglines are not clear
- [ ] Fix the upgrade job behavior to avoid tripping `CannotRetrieveUpdates` alert

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64 periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64 periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ovn-remote-libvirt-ppc64le

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64

petr-muller commented:
/hold cancel

petr-muller commented:
/pj-rehearse pull-ci-maistra-envoy-maistra-2.4-maistra-envoy-unit-2-4 periodic-ci-dora-metrics-pelorus-master-4.13-e2e-openshift-test-scenario-1-periodic pull-ci-opendatahub-io-kubeflow-master-odh-notebook-controller-e2e pull-ci-3scale-3scale-operator-master-test-e2e pull-ci-openshift-coredns-master-e2e-aws-ovn pull-ci-medik8s-machine-deletion-remediation-main-4.14-openshift-e2e periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-gcp-ovn-upgrade periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-aws-sdn-arm64

dgoodwin commented:
/approve

petr-muller commented:
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.14-ocp-e2e-ibmcloud-ovn-multi-s390x

petr-muller commented:
/pj-rehearse ack
